docs/concepts/validators.md

```diff
@@ -1,635 +1,575 @@
+        from typing_extensions import Annotated
+
+        from pydantic import BaseModel, Field, ValidationError, ValidatorFunctionWrapHandler, field_validator
+
+
+        class Model(BaseModel):
+            my_string: Annotated[str, Field(max_length=5)]
 
-!!! note "On Inheritance"
-    A `@model_validator` defined in a base class will be called during the validation of a subclass instance.
+            @field_validator('my_string', mode='wrap')
+            @classmethod
+            def truncate(cls, value: Any, handler: ValidatorFunctionWrapHandler) -> str:  # (1)!
+                try:
+                    return handler(value)
+                except ValidationError as err:
+                    if err.errors()[0]['type'] == 'string_too_long':
+                        return handler(value[:5])
+                    else:
+                        raise
 
-    Overriding a `@model_validator` in a subclass will override the base class' `@model_validator`, and thus only the subclass' version of said `@model_validator` will be called.
 
-Model validators can be `mode='before'`, `mode='after'` or `mode='wrap'`.
+        print(Model(my_string='abcde'))
+        #> my_string='abcde'
+        print(Model(my_string='abcdef'))
+        #> my_string='abcde'
+        ```
 
-Before model validators are passed the raw input which is often a `dict[str, Any]` but could also be an instance of the model itself (e.g. if `UserModel.model_validate(UserModel.construct(...))` is called) or anything else since you can pass arbitrary objects into `model_validate`.
-Because of this `mode='before'` validators are extremely flexible and powerful but can be cumbersome and error prone to implement.
-Before model validators should be class methods.
-The first argument should be `cls` (and we also recommend you use `@classmethod` below `@model_validator` for proper type checking), the second argument will be the input (you should generally type it as `Any` and use `isinstance` to narrow the type) and the third argument (if present) will be a `pydantic.ValidationInfo`.
+        1. Unlike the other validator types, *wrap* validators must be defined with a second `handler` argument,
+           even if they don't make use of it.
 
-`mode='after'` validators are instance methods and always receive an instance of the model as the first argument. Be sure to return the instance at the end of your validator.
-You should not use `(cls, ModelType)` as the signature, instead just use `(self)` and let type checkers infer the type of `self` for you.
-Since these are fully type safe they are often easier to implement than `mode='before'` validators.
-If any field fails to validate, `mode='after'` validators for that field will not be called.
+!!! note "Validation of default values"
+    As mentioned in the [fields documentation](./fields.md#validate-default-values), default values of fields
+    are *not* validated by default, and thus custom validators will not be applied as well.
```

Comment:
```suggestion
    are *not* validated by default, and thus custom validators are also not applied to default values by default.
```

This is a lot of "default default default", is there a better way we can phrase this?

Reply:
Rephrased to _default values of fields are not validated unless configured to do so_ (and the crosslink can be used to check how to do so).